resampler/src/speex.rs

```diff
@@ -1489,12 +1482,10 @@ fn speex_resampler_process_native(
     *out_len = out_sample as u32;
     st.last_sample[channel_index as usize] -= *in_len;
     let ilen: u32 = *in_len;
-    let mut j: i32 = 0;
-    while j < n - 1 {
+    (0..n - 1).for_each(|j| {
         st.mem[mem_idx + j as usize] =
             st.mem[mem_idx + (j as u32 + ilen) as usize];
```

Comment:
i tried to replace this with enumerating on `st.mem` but failed because its borrowed twice here :man_shrugging:

Reply:
Since the second `st.mem` remains the same, we could clone that slice. In this way we don't borrow twice. Something like:
`let mem_clone = &st.mem[mem_idx + n + ilen].clone()`